pkg/config/config.go

```diff
@@ -99,24 +99,60 @@ func (c *Config) FindSiteByHostName(hostname string) (*Site, error) {
 	return nil, fmt.Errorf("unable to find site with hostname %s", hostname)
 }
 
-// ListOfSitesByDirectory takes the users home directory and the current
-// working directory and returns a list of sites that are "context-aware".
+// ListOfSitesByDirectory takes the user’s home directory and the current
+// working directory and returns a list of sites within that context.
 func (c *Config) ListOfSitesByDirectory(home, wd string) []Site {
 	var found []Site
+
+	// get any sites whose container paths are within the working directory
 	for _, s := range c.Sites {
-		p, _ := s.GetAbsPath(home)
+		p, _ := s.GetAbsContainerPath(home)
 
-		// check if the path matches a sites path, then we are in a known site
-		if strings.Contains(wd, p) {
+		if strings.Contains(p, wd) {
 			found = append(found, s)
 		}
 	}
 
-	// if we found any matching sites, return those
+	// if we found a subset of sites, let’s see if we can narrow and return it
 	if len(found) > 0 {
+		// require at least one segment match if we’re being more specific
```

Comment:
You have great test coverage, but I'm wondering if the logic can be cleaned up here while keeping the logic intact?

Reply:
Slightly improved my logic here and hopefully made it more readable!